compiler/resolution/initializerResolution.cpp

```diff
@@ -554,33 +559,39 @@ static void resolveInitCall(CallExpr* call, AggregateType* newExprAlias, bool fo
           // In the future, the compiler should not be attempting to resolve
           // an already-resolved call.
           bool existingErrors = fatalErrorsEncountered();
-          if (newExprAlias != NULL) {
+          if (newExprAlias != NULL && emitErrors) {
             USR_FATAL_CONT(call, "Unable to resolve new-expression with type alias '%s'", newExprAlias->symbol->name);
           }
           if (!inGenerousResolutionForErrors()) {
             startGenerousResolutionForErrors();
-            resolveInitCall(call, newExprAlias, /*forNewExpr*/ false);
+            const bool forNewExpr = false;
+            resolveInitCall(call, emitErrors, newExprAlias, forNewExpr);
             FnSymbol* retry = call->resolvedFunction();
             stopGenerousResolutionForErrors();
 
             if (fIgnoreNilabilityErrors && existingErrors == false && retry)
               clearFatalErrors();
           }
         } else {
-          if (candidates.n == 0) {
-            printResolutionErrorUnresolved(info, mostApplicable);
-
-            USR_STOP();
+          if (emitErrors) {
+            if (candidates.n == 0) {
+              printResolutionErrorUnresolved(info, mostApplicable);
+
+              USR_STOP();
+            } else {
+              printResolutionErrorAmbiguous (info, candidates);
+            }
           } else {
-            printResolutionErrorAmbiguous (info, candidates);
+            INT_ASSERT(tryingToResolve());
```

Comment:
I'm not sure how I feel about this - if we shouldn't emit errors, why is turning this into an internal compiler error okay?

Reply:
Good point, this assertion is a holdover from before the `bool` formal.